### ISSUE TITLE
Decouple vt100.js from main project

### ISSUE THREAD

**Issue by KLuka** [shellinabox/shellinabox#280]
_From [kveron...@gmail.com](https://code.google.com/u/110579634229550218345/) on February 15, 2015 01:01:44_

I used the full ShellInABox project for awhile, and always loved how nicely accurate the VT100 Terminal Emulation is, and how well it works on multiple platforms.  I used it for a terminal-based based game last year, but found that the scalability of using ShellInABox for such a project wasn't good...

This lead me to rewrite the game engine in WebSockets, and originally I used jquery.terminal as the Terminal...  It was very lackluster and didn't support any VT100 control commands.  As my game is growing, I am in need of full VT100 emulation again, and so I have once again turned to ShellInABox, but not the server-side component.  I have to the best of my ability decoupled vt100.js from ShellInABox.  Unfortunately using vt100.js in a standalone project isn't provided anywhere, and only a demo.js was slightly helpful.

So, I'd like to see vt100.js spin into it's own project, so that it can be used without ShellInABox.  Here's a link to my terminal-based game that uses vt100.js, as an example: http://www.hackers-edge.com/play/ At the time of this writing, the game is currently in closed beta, however I'd be glad to give you beta access just for creating such a great VT100 terminal emulator:  ABBHF (Invite code)

_Original issue: http://code.google.com/p/shellinabox/issues/detail?id=280_

**Comment by kveroneau** [comment]
Is there any update on this now that the project is successfully migrated over to Git?  I still use a version on the above URL.  It would also be nice to have some documentation on vt100.js as well, such as what VT100/ANSI control codes work, and which ones don't.  I am also curious on how the mouse support works, from my understanding, this is an extension of sorts from xterm:  http://www.xfree86.org/4.7.0/ctlseqs.html  Although I am still trying to grasp how to enable support in my Telnet application.  I know I am sort of using vt100.js in a rather unsupported way, but by far, this is one of the best terminal emulators for the web out there.  I highly recommend anybody curious on how I used it in my application here to check out my client.js file.  My application also uses WebSockets over primitive AJAX to communicate the backend service.  I highly recommend that ShellInABox eventually add a WebSockets option as well in the future, so say shellinabox --websockets would enable that backend over traditional Ajax polling.

**Comment by uudruid74** [comment]
The simpler mouse controls work.  You can try xterm.js.   LOL ... I like your choice of port numbers for the game.   Do you have your fork of vt100.js on a github somewhere?   I was thinking of webrtc rather than websockets for the basic data stream and would like to really clean up some of the mess - like get rid of the built-in web server completely and run it as a fastcgi process (using ajax) as a fallback when webrtc isn't supported .. basically I think it needs to go on the chopping block!

**Comment by kveroneau** [comment]
I don't believe I forked vt100.js, only used the JavaScript class it exposes in client.js.  You can do a diff against your current version if you'd like:  http://www.hackers-edge.com/s/js/vt100.js

Client.JS can be found here to see how I used it in my own code: http://www.hackers-edge.com/s/js/client.js

I did manage to get mouse support working, it uses standard ANSI escape codes which also work on the standard OpenBSD Telnet client in Konsole and Xterm.
